os/board/common/crashdump.c

```diff
@@ -68,39 +73,45 @@ static int ramdump_via_uart(void)
 	size_t size;
 	uint8_t *ptr;
 	char host_reg[1] = "";
-	char *target_str = HANDSHAKE_STRING;
+	char *target_str = RAMDUMP_HANDSHAKE_STRING;
 	char host_buf[HANDSHAKE_STR_LEN_MAX] = "";
+	uint32_t userfs_start;
 
 #if !defined(CONFIG_ARCH_LOWPUTC)
-	/* If lowlevel serial is not available, ramdump is not possible */
+	/* If lowlevel serial is not available, dumping is not possible */
 	return -1;
 #endif
 
 	/* Inform the terminal user */
 	up_puts("\n************************************************************************************************\n");
-	up_puts("\t\tDisconnect this serial terminal and Run Ramdump Tool\n");
+	up_puts("\t\tDisconnect this serial terminal and run Dump Tool\n");
 	up_puts("************************************************************************************************\n");
 
 	/* Receive hanshake string from HOST */
 	do {
 		host_buf[0] = up_getc();
 	} while (host_buf[0] != target_str[0]);
 
-	for (i = 1; i < strlen(target_str);) {
+	for (i = 1; i < HANDSHAKE_STR_LEN_MAX; ) {
 		if ((ch = up_getc()) != -1) {
 			host_buf[i] = ch;
 			i++;
 		}
 	}
 
-	if (strncmp(host_buf, target_str, strlen(target_str)) != 0) {
-		/* Send NAK */
-		up_lowputc('N');
+	/* Check handshake string against RAMDUMP HANDSHAKE string */
+	if (strncmp(host_buf, RAMDUMP_HANDSHAKE_STRING, HANDSHAKE_STR_LEN_MAX) == 0) {
+		up_lowputc('A');
+	} else {
+		if (strncmp(host_buf, FSDUMP_HANDSHAKE_STRING, HANDSHAKE_STR_LEN_MAX) == 0) {
```

Comment:
You'd better compare with `strlen(FSDUMP_HANDSHAKE_STRING) + 1` instead of `HANDSHAKE_STR_LEN_MAX` for security.